test.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -eux
+
+# Prepare env vars
+OS=${OS:="centos"}
+OS_VERSION=${OS_VERSION:="7"}
+PYTHON_VERSION=${PYTHON_VERSION:="2"}
+IMAGE="$OS:$OS_VERSION"
+# Pull fedora images from registry.fedoraproject.org
+if [[ $OS == "fedora" ]]; then
+  IMAGE="registry.fedoraproject.org/$IMAGE"
+fi
+
+CONTAINER_NAME="atomic-reactor-$OS-$OS_VERSION-py$PYTHON_VERSION"
+RUN="docker exec -ti $CONTAINER_NAME"
+if [[ $OS == "fedora" ]]; then
+  if [[ $OS_VERSION == 25 && $PYTHON_VERSION == 2 ]]; then PIP_PKG="python-pip"; else PIP_PKG="python$PYTHON_VERSION-pip"; fi
+  PIP="pip$PYTHON_VERSION"
+  PKG="dnf"
+  PKG_EXTRA="dnf-plugins-core"
+  BUILDDEP="dnf builddep"
+  PYTHON="python$PYTHON_VERSION"
+  PYTEST="py.test-$PYTHON_VERSION"
+else
+  PIP_PKG="python-pip"
+  PIP="pip"
+  PKG="yum"
+  PKG_EXTRA="yum-utils epel-release git-core"
+  BUILDDEP="yum-builddep"
+  PYTHON="python"
+  PYTEST="py.test"
+fi
+# Create container if needed
+if [[ $(docker ps -q -f name=$CONTAINER_NAME | wc -l) -eq 0 ]]; then
+  docker run --name $CONTAINER_NAME -d -v $PWD:$PWD -w $PWD -ti $IMAGE sleep infinity
+fi
+
+# Install dependencies
+$RUN $PKG install -y $PKG_EXTRA
+$RUN $BUILDDEP -y atomic-reactor.spec
+if [[ $OS != "fedora" ]]; then
+  # Install dependecies for test, as check is disabled for rhel
+  $RUN yum install -y python-flexmock python-six \
+                      python-docker-py python-backports-lzma \
+                      python-responses \
+                      python-requests python-requests-kerberos # OSBS dependencies
+fi
+
+# Install package
+$RUN $PKG install -y $PIP_PKG
+if [[ $PYTHON_VERSION == 3 && $OS_VERSION == rawhide ]]; then
+  # https://fedoraproject.org/wiki/Changes/Making_sudo_pip_safe
+  $RUN mkdir -p /usr/local/lib/python3.6/site-packages/
+fi
+
+# Install osbs-client and docker_squash
+$RUN $PKG install -y osbs-client
+$RUN $PIP install --upgrade --no-deps --force-reinstall git+https://github.com/projectatomic/osbs-client
```

Comment:
Why do we need to install osbs-client twice?
Seems to me that the `requirements.txt` file is not being processed at all anymore.

Reply:
This is required to a) setup all deps for osbs-client from packages and b) update osbs-client from latest master without attempting to install deps from PIP.

This is done mostly for docker-squash, which tries to pull latest docker-py and fails as there is already one installed from RPM. Other deps might want gcc etc.

Reply:
Right, but it means that if we add a new dependency to osbs-client, it won't be installed until it's actually released for the underlying OS.

Reply:
Yes, the `test.sh` packages in `PKG_EXTRA` with specific packages names for CentOS/Fedora version, as installing deps from pip would take too much time (most of these require `gcc` etc.)

Reply:
@twaugh, thoughts?

Reply:
The way to resolve this:
* pull latest code from osbs-client master branch
* install the deps from the spec
* install fresh osbs-client via `pip --upgrade --no-deps --force-reinstall`

This would ensure the latest deps are recorded in the spec of osbs-client

Reply:
@vrutkovs, that looks like a good way to do it IMHO.

Reply:
Implemented that, PTAL